datafusion/common/src/config.rs

```diff
@@ -470,7 +478,7 @@ config_namespace! {
         /// delta_byte_array, rle_dictionary, and byte_stream_split.
         /// These values are not case sensitive. If NULL, uses
         /// default parquet writer setting
-        pub encoding: Option<String>, default = None
+        pub encoding: Option<String>, transform = str::to_lowercase, default = None
```

Comment:
I feel it would be useful to describe the transform field in this case and other cases where `transform` has been introduced

Reply:
Feels like "These values are not case sensitive" is a good indication? It wasn't always true before but it is now

Reply:
What I mean is since the new field `transform` introduced, it is good to highlight how this field is gonna be used.
Previously the config structs were the same format having a config field and a default value and it was self-explanatory. Having introduced the `transform` I feel its needed to comment how its gonna be used

Reply:
Do you think this would be enough? https://github.com/apache/datafusion/pull/13576/files#diff-2f697c310af1a48521829d8bd665cf64b6415fbf88edd370efa30b1ed686354dR42-R73

Reply:
Nice, I applied some formatting to highlight the `transform`

Reply:
thank you! do you want to approve?